R/utils-cache.R

```diff
@@ -0,0 +1,57 @@
+#' Standardize text for hashing
+#'
+#' Make sure text after styling results in the same hash as text before styling
+#' if it is indeed identical.
+#' @param text A character vector.
+#' @keywords internal
+hash_standardize <- function(text) {
+  text <- ensure_last_is_empty(text)
+  Encoding(text) <- "UTF-8"
+  list(text)
+}
+
+#' Where is the cache?
+#'
+#' Finds the path to the cache and creates it if it does not exist.
+#' @inheritParams cache_clear
+#' @keywords internal
+cache_find_path <- function(cache_name = NULL) {
+  cache_name <- cache_get_or_derive_name(cache_name)
+  R.cache::getCachePath(c("styler", cache_name))
+}
+
+#' Check if a cache is activated
+#'
+#' @param cache_name The name of the cache to check. If `NULL`, we check if
+#'   any cache is activated. If not `NULL`, we check if a specific cache is
+#'   activated.
+#' @keywords internal
+cache_is_activated <- function(cache_name = NULL) {
+  current_cache <- cache_get_name()
+  if (is.null(cache_name)) {
+    !is.null(current_cache)
+  } else if (!is.null(current_cache)) {
+    cache_name == current_cache
+  } else {
+    FALSE
+  }
+}
+
+cache_derive_name <- function() {
+  utils::packageDescription("styler", fields = "Version")
```

Comment:
Can we derive without looking at the package library? Maybe store in a global variable, this is evaluated during installation?

Reply:
Not during installation, but `.onLoad()`. The cache is version specific because a new version of styler potentially styles code differently, so I need the version number and I don't know what a better API would be to it. Suggestions?

Reply:
If you write `v <- read.dcf("../DESCRIPTION")[...]` in an R file, I think it's evaluated when the package is built. Can you please double-check?

Reply:
`v <- read.dcf("DESCRIPTION")` seems to work locally yes. Quick Googling did not help. However, if it is brittle, I think I'd prefer the other approach, it's a detail. I implemented it in a2be25905ceee35e4f780a4f7b1cbbbfe8e3d0f7 and left the function call. Alternative would be to directly use a variable `version` instead of the call that derives it.

Reply:
I feel it's more robust than looking at the library, because the version in your library might be different from the version loaded. Better use `version <- read.dcf(...)[, "Version"]` though.

Reply:
ok